### PR TITLE
looking up for approval instituion in csv

### PR DESCRIPTION
**Pull request by truhacevkir**
Provided csv contains all departments that have been transferred from one institution to another.

When we do not find approval for creator in Cristin approvals based on institution identifier, we look up in csv. 
If csv contains transfer between creator institution and one of institutions from the Cristin approvals, we "connect" following institutions and creating DbInstitutionPoints from them.

### CODE REVIEW

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java [hunk 1]
```diff
@@ -289,13 +295,43 @@ private static BigDecimal extractAuthorPointsForAffiliation(ScientificPerson sci
         return new BigDecimal(scientificPerson.getAuthorPointsForAffiliation());
     }
 
-    private static URI getTopLevelOrganization(ScientificPerson scientificPerson, List<CristinLocale> institutions) {
+    private URI getTopLevelOrganization(ScientificPerson scientificPerson, List<CristinLocale> institutions) {
+        var list = institutions.stream()
```
Comment (monaullah) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java:299]: Rename var to something more describing than `list`?

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinNviReportEventConsumer.java [hunk 2]
```diff
@@ -102,4 +113,17 @@ private CandidateDao createAndPersist(CristinNviReport cristinNviReport) {
             return repository.create(candidate, approvals, yearReported);
         }
     }
+
+    private List<CristinDepartmentTransfer> readCristinDepartments() {
+        try {
+            MappingIterator<CristinDepartmentTransfer> iterator =
```
Comment (monaullah) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinNviReportEventConsumer.java:119]: Codacy issue: Ensure that resources like this StringReader object are closed after use

**Review by monaullah — CHANGES_REQUESTED** [review]

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java (hunk 1 above)
Comment (monaullah) [reply event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java:299]: Maybe findFirst() is what you want here?

Comment (truhacevkir) [reply event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java:299]: Refactoring to:

```
    private URI getTopLevelOrganization(ScientificPerson scientificPerson, List<CristinLocale> institutions) {
        return institutions.stream()
                             .filter(cristinLocale -> hasSameInstitutionIdentifier(scientificPerson, cristinLocale))
                             .map(CristinMapper::constructInstitutionId)
                             .findFirst().orElseGet(() -> extractInstitutionIdentifierForTransferredInstitution(scientificPerson, institutions));
    }
```

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinNviReportEventConsumer.java (hunk 2 above)
Comment (truhacevkir) [reply event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinNviReportEventConsumer.java:120]:
```
    private List<CristinDepartmentTransfer> readCristinDepartments() {
        try (StringReader reader = new StringReader(CRISTIN_DEPARTMENT_TRANSFERS_STRING)) {
            MappingIterator<CristinDepartmentTransfer> iterator =
                new CsvMapper()
                    .readerFor(CristinDepartmentTransfer.class)
                    .with(CsvSchema.emptySchema().withHeader())
                    .readValues(reader);
            return iterator.readAll();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
```

**Review by monaullah — APPROVED** [review]

**Pull request merged**